lib/checkPullRequestTemplate.js

```diff
@@ -0,0 +1,237 @@
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkPRExplanation = (pullRequest) => {
+  /*
+    A valid overview section should look like:
+    ## Overview
+    1. This PR fixes or fixes part of #[fill_in_number_here].
+    2. This PR does the following: This PR does a bunch of things.
+
+    An invalid overview section would look like:
+    ## Overview
+    1. This PR fixes or fixes part of #[fill_in_number_here].
+    2. This PR does the following: [Explain here what your PR does and why].
+  */
+
+  // Restrict search to the overview section.
+  const overviewText = '## Overview';
+  const overviewHeadingIndex = pullRequest.body.indexOf(overviewText);
+
+  // Checks that there is an overview.
+  if (overviewHeadingIndex !== -1) {
+    // Get the index of the next heading after the overview.
+    const nextHeadingIndex = pullRequest.body.indexOf(
+      '## ',
+      overviewHeadingIndex + overviewText.length
+    );
+
+    // The overview section is the section between the overview heading
+    // and another heading (usually the essential checklist heading).
+    const overviewSection = pullRequest.body.substring(
+      overviewHeadingIndex,
+      nextHeadingIndex
+    );
+
+    // The PR explanaion is the second item in the overview section hence,
+    // should start with "2".
+    let explanation = overviewSection.split('\n').find((line) => {
+      return line.startsWith(EXPLANATION_START);
+    });
+
+    if (explanation.includes(EXPLANATION_TEXT)) {
+      // The user has not added the explanation.
+      return (
+        'the body of this PR is missing the required description, ' +
+        'please update the body with a description of what this PR does.'
+      );
+    }
+    return '';
+  } else {
+    // There is no overview section in the body of the PR.
+    return (
+      'the body of this PR is missing the overview section, please ' +
+      'update it to include the overview.'
+    );
+  }
+};
+
+/**
+ * Checks that the pull request's body contains the explanation.
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const checkEssentialChecklist = (pullRequest) => {
+  /*
+    A valid checklist section would look like:
+    ## Essential Checklist
+    - [ ] The PR title starts with "Fix #bugnum: ", followed by a short
+    , clear summary of the changes. (If this PR fixes part of an issue,
+    prefix the title with "Fix part of #bugnum: ...".)
+    - [x] The linter/Karma presubmit checks have passed locally on
+    your machine.' +
+    - [x] "Allow edits from maintainers" is checked. (See [here]
+    (https://help.github.com/en/github/collaborating-with-issues-and
+    -pull-requests/allowing-changes-to-a-pull-request-branch-created-
+    from-a-fork) for instructions on how to enable it.)  - This
+    lets reviewers restart your CircleCI tests for you.
+    - [ ] The PR is made from a branch that's **not** called
```

Comment:
This should be checked.

Reply:
Okay, but we are not checking that here.